src/seq.rs

```diff
@@ -475,49 +480,97 @@ impl PortSubscribe {
 
 }
 
+#[derive(Clone)]
+enum EventExtraData<'a> {
+    None,
+    Borrowed(&'a [u8]),
+    Owned(Vec<u8>)
+}
+
+impl<'a> EventExtraData<'a> {
+    fn into_owned<'b>(self) -> EventExtraData<'b> {
+        match self {
+            EventExtraData::None => EventExtraData::None,
+            EventExtraData::Borrowed(buf) => EventExtraData::Owned(buf.to_vec()),
+            EventExtraData::Owned(vec) => EventExtraData::Owned(vec)
+        }
+    }
+}
+
 /// [snd_seq_event_t](http://www.alsa-project.org/alsa-doc/alsa-lib/structsnd__seq__event__t.html) wrapper
 ///
 /// Fields of the event is not directly exposed. Instead call `Event::new` to set data (which can be, e g, an EvNote).
 /// Use `get_type` and `get_data` to retreive data.
-pub struct Event(alsa::snd_seq_event_t, EventType, Option<Vec<u8>>);
+///
+/// The lifetime parameter refers to the lifetime of an associated external buffer that might be used for
+/// variable-length messages (e.g. SysEx).
+pub struct Event<'a>(alsa::snd_seq_event_t, EventType, EventExtraData<'a>);
 
-unsafe impl Send for Event {}
+unsafe impl<'a> Send for Event<'a> {}
 
-impl Event {
-    pub fn new<D: EventData>(t: EventType, data: &D) -> Self {
-        let mut z = Event(unsafe { mem::zeroed() }, EventType::None, None);
+impl<'a> Event<'a> {
+    pub fn new<D: SetEventData>(t: EventType, data: D) -> D::Output {
+        let mut z = Event(unsafe { mem::zeroed() }, EventType::None, EventExtraData::None);
         z.1 = t;
         (z.0)._type = t as c_uchar;
+        let length_flags = match t {
+            EventType::Sysex => SND_SEQ_EVENT_LENGTH_VARIABLE,
+            EventType::Bounce => SND_SEQ_EVENT_LENGTH_VARIABLE, // not clear whether this should be VARIABLE or VARUSR
+            EventType::UsrVar0 => SND_SEQ_EVENT_LENGTH_VARUSR,
+            EventType::UsrVar1 => SND_SEQ_EVENT_LENGTH_VARUSR,
+            EventType::UsrVar2 => SND_SEQ_EVENT_LENGTH_VARUSR,
+            EventType::UsrVar3 => SND_SEQ_EVENT_LENGTH_VARUSR,
+            EventType::UsrVar4 => SND_SEQ_EVENT_LENGTH_VARUSR,
+            _ => SND_SEQ_EVENT_LENGTH_FIXED
+        };
+        (z.0).flags |= length_flags;
         debug_assert!(D::has_data(t));
-        data.set_data(&mut z);
-        z
+        data.set_data(z)
+    }
+
+    /// Consumes this event and returns an (otherwise unchanged) event where the externally referenced
+    /// buffer for variable length messages (e.g. SysEx) has been copied into the event.
+    /// The returned event has an arbitrary lifetime that is decoupled from the original buffer.
+    pub fn into_owned<'b>(self) -> Event<'b> {
```

Comment:
I agree that it should return `Event<'static>`, but the name `into_static` has no precedence in the standard library or the [Rust naming conventions](https://aturon.github.io/style/naming.html). I'm not totally happey with `into_owned`, either, but I'd prefer it over `into_static`.